mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -82,6 +82,19 @@ object MLUtils {
           val value = indexAndValue(1).toDouble
           (index, value)
         }.unzip
+
+        // check if indices is one-based and in ascending order
```

Comment:
1. I will use `require` for the check.
2. indices is -> indices are
And since there's a `-1` in line 81, it's acceptable to have indices(0) == 0 here. (changed to zero-based). 
The first ut has covered the check for zero-based LIBSVM file.
Let me know if I should change more. Thanks a lot for the careful check.